tests/test_config.py

```diff
@@ -38,10 +36,11 @@ def test_load_ssl_config_verify_env_file(https_server, ca_cert_pem_file, config)
         else str(Path(ca_cert_pem_file).parent)
     )
     ssl_config = SSLConfig(trust_env=True)
```

Comment:
It'd be *great* as part of this change if we dropped using the private `SSLConfig` from the tests completely.
Not *necessarily* a blocker to getting this PR in, but might be worth a go?